### PR TITLE
ASCN-242: create release on merge to main

### PR DESCRIPTION
**Pull request by stephen-vakil**
This configures automated release+deploy to dev upon merge to main.  As a bonus, it also configures automated release+deploy to sandbox if the pr has a label `deploy-to-sandbox`.

# How to test
Check the action history and verify in octopus that it deployed that version.

### CODE REVIEW

**Comment by mud5150** [comment]
Seems reasonable. It may be advantageous to use the first class [Octopus actions](https://octopus.com/docs/packaging-applications/build-servers/github-actions) rather than a script, but it looks like everything should work for what we need. The published actions look like they handle looking up the projects and environments by name so we don't have to copy around the IDs.

**Review by mud5150 — APPROVED** [review]

File: .github/workflows/workflow_build_and_release_containers.yaml [hunk 1]
```diff
@@ -42,4 +43,39 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install octopus .NET tool
+        run: |
+          dotnet tool install --global Octopus.DotNet.Cli --version 9.1.7
+          dotnet octo --version
+      - name: determine if we should create release and deploy
+        id: should-create-release
+        run: |
+          if [[ ${{ github.ref_name == 'main' || github.pull_request.labels.*.name == 'deploy-to-sandbox' }} ]]; then
```
Comment (aalear) [inline .github/workflows/workflow_build_and_release_containers.yaml:58]: I don't want to hold this PR up on this point, but I don't know how I feel about automatic sandbox deployments based on a label. It sounds like a nice quality-of-life convenience feature for an app like Opserver that changes incredibly rarely, but I don't think I'd want to replicate it in other apps where the odds of developers stepping on each other's toes by continually redeploying their PRs to the sandbox on every commit are much higher.

Comment (stephen-vakil) [reply .github/workflows/workflow_build_and_release_containers.yaml:58]: if that happens we could just undo this / turn this off until we can create per-pr environments, right?

Comment (mud5150) [reply .github/workflows/workflow_build_and_release_containers.yaml:58]: If we wanted to use sandbox long term for branch testing, I would say to namespace based on the branch or PR with separate instances and ingress rules so PRs don't clobber each other.

Comment (stephen-vakil) [reply .github/workflows/workflow_build_and_release_containers.yaml:58]: I am going to merge as is for now - let's keep the discussion going in slack and make any changes in a future pr.  by merging now I can validate that the dev merge works :)

**Pull request merged**